x/wasm/keeper/handler_plugin_encoders.go

```diff
@@ -327,33 +326,6 @@ func EncodeIBCMsg(portSource types.ICS20TransferPortSource) func(ctx sdk.Context
 				Memo:             msg.Transfer.Memo,
 			}
 			return []sdk.Msg{msg}, nil
-		case msg.PayPacketFee != nil:
-			fee, err := ConvertIBCFee(&msg.PayPacketFee.Fee)
-			if err != nil {
-				return nil, errorsmod.Wrap(err, "fee")
-			}
-			msg := &ibcfeetypes.MsgPayPacketFee{
-				Fee:             fee,
-				SourcePortId:    msg.PayPacketFee.PortID,
-				SourceChannelId: msg.PayPacketFee.ChannelID,
-				Signer:          sender.String(),
-				Relayers:        msg.PayPacketFee.Relayers,
-			}
-			return []sdk.Msg{msg}, nil
-		case msg.PayPacketFeeAsync != nil:
-			fee, err := ConvertIBCFee(&msg.PayPacketFeeAsync.Fee)
-			if err != nil {
-				return nil, errorsmod.Wrap(err, "fee")
-			}
-			msg := &ibcfeetypes.MsgPayPacketFeeAsync{
-				PacketId: channeltypes.PacketId{
-					PortId:    msg.PayPacketFeeAsync.PortID,
-					ChannelId: msg.PayPacketFeeAsync.ChannelID,
-					Sequence:  msg.PayPacketFeeAsync.Sequence,
-				},
-				PacketFee: ibcfeetypes.NewPacketFee(fee, sender.String(), msg.PayPacketFeeAsync.Relayers),
-			}
-			return []sdk.Msg{msg}, nil
```

Comment:
I would suggest keeping the cases here and add more specific error messages than the fallback

Reply:
Good point. Added that and tests for it too.